server-timing/load.php

```diff
@@ -113,3 +116,72 @@ function perflab_wrap_server_timing( $callback, $metric_slug, $access_cap ) {
 		return $result;
 	};
 }
+
+/**
+ * Registers the Server-Timing setting.
+ *
+ * @since n.e.x.t
+ */
+function perflab_register_server_timing_setting() {
+	register_setting(
+		PERFLAB_SERVER_TIMING_SCREEN,
+		PERFLAB_SERVER_TIMING_SETTING,
+		array(
+			'type'              => 'object',
+			'sanitize_callback' => 'perflab_sanitize_server_timing_setting',
+			'default'           => array(),
+		)
+	);
+}
+add_action( 'init', 'perflab_register_server_timing_setting' );
+
+/**
+ * Sanitizes the Server-Timing setting.
+ *
+ * @since n.e.x.t
+ *
+ * @param mixed $value Server-Timing setting value.
+ * @return array Sanitized Server-Timing setting value.
+ */
+function perflab_sanitize_server_timing_setting( $value ) {
+	if ( ! is_array( $value ) ) {
+		return array();
+	}
+
+	/*
+	 * Ensure that every element is an indexed array of hook names.
+	 * Any duplicates across a group of hooks are removed.
+	 */
+	return array_filter(
+		array_map(
+			static function( $hooks ) {
+				if ( ! is_array( $hooks ) ) {
+					$hooks = explode( "\n", $hooks );
+				}
+				return array_values(
+					array_unique(
+						array_filter(
+							array_map(
+								static function( $hookname ) {
+									/*
+									 * Allow any characters except whitespace.
+									 * While most hooks use a limited set of characters, hook names in plugins are not
+									 * restricted to them, therefore the sanitization does not limit the characters
+									 * used.
+									 */
+									return preg_replace(
+										'/\s/',
+										'',
+										sanitize_text_field( $hookname )
+									);
+								},
+								$hooks
+							)
+						)
+					)
+				);
+			},
+			$value
+		)
```

Comment:
There is a lot of nesting happening here. Not a problem, but just makes it a bit hard on the eyes. Not a blocker.

Reply:
Reduced the nesting a bit in https://github.com/WordPress/performance/pull/784/commits/7f99021dd4afe5420dc126a96b6d26a8ea0bf16c